src/Type/PostObject/Mutation/PostObjectDelete.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace WPGraphQL\Type\PostObject\Mutation;
+
+use GraphQLRelay\Relay;
+use WPGraphQL\Types;
+
+/**
+ * Class PostObjectDelete
+ *
+ * @package WPGraphQL\Type\PostObject\Mutation
+ */
+class PostObjectDelete {
+
+	/**
+	 * Holds the mutation field definition
+	 *
+	 * @var array $mutation
+	 */
+	private static $mutation;
+
+	/**
+	 * Defines the delete mutation for PostTypeObjects
+	 *
+	 * @param \WP_Post_Type $post_type_object
+	 *
+	 * @return array|mixed
+	 */
+	public static function mutate( \WP_Post_Type $post_type_object ) {
+
+		if ( null === self::$mutation ) {
+			self::$mutation = [];
+		}
+
+		if ( ! empty( $post_type_object->graphql_single_name ) && empty( self::$mutation[ $post_type_object->graphql_single_name ] ) ) :
+
+			/**
+			 * Set the name of the mutation being performed
+			 */
+			$mutation_name = 'delete' . ucwords( $post_type_object->graphql_single_name );
+
+			self::$mutation[ $post_type_object->graphql_single_name ] = Relay::mutationWithClientMutationId( [
+				'name'                => esc_html( $mutation_name ),
```

Comment:
So this is the name in the context of the Schema, not in the context of executing, so I think esc_html is still appropriate as it gets sent to the client, not the server